src/pps/filter.go

```diff
@@ -0,0 +1,22 @@
+package pps
+
+import (
+	context "context"
+)
+
+func (r *ListDatumRequest_Filter) Allow(ctx context.Context, item *DatumInfo) bool {
+	if r == nil {
+		return true
+	}
+	match := false
+	for _, s := range r.State {
+		if s == item.State {
+			match = true
```

Comment:
You could just return true here and `len(r.State) == 0` below.

Reply:
I wanted to leave the right groundwork for multiple attributes, but you’re probably right: doing the simple thing now makes sense.